### ISSUE TITLE
Extract method / functiom must give feedback when a selection is not extractable and why

### ISSUE THREAD

**Issue by dbaeumer** [Microsoft/vscode#33501]
Testing #33282
```ts
function foo() {
	for (let i = 0; i < 10; i++) {
		if (i === 5)
			return;
	}
	console.log('Should not be printed');
}
```
- select
```ts
	for (let i = 0; i < 10; i++) {
		if (i === 5)
			return;
	}
```
- try to extract
- nothing happens. No UI feedback why
- change code to:
```ts
	for (let i = 0; i < 10; i++) {
		return;
	}
```
- try extract again.

Observe: not it works.

Unclear if TS server provides info on why this is not possible in the first example.

**Comment by mjbvz** [comment]
I agree this would be nice but it clashes with our current code action based refactoring model. If we ever move to a model where the TS extension asks TS Server for a specific refactoring, we should include this as part of the design
